Jenkinsfile

```diff
@@ -0,0 +1,65 @@
+def gitCommit() {
+    sh "git rev-parse HEAD > GIT_COMMIT"
+    def gitCommit = readFile('GIT_COMMIT').trim()
+    sh "rm -f GIT_COMMIT"
+    return gitCommit
+}
+
+def gitEmail() {
+    sh "git --no-pager show -s --format='%ae' ${gitCommit()} > GIT_EMAIL"
+    def gitEmail = readFile('GIT_EMAIL').trim()
+    sh "rm -f GIT_EMAIL"
+    return gitEmail
+}
+
+def ipAddress() {
+    sh "docker inspect test-container-${env.BUILD_NUMBER} | jq -r '.[0].NetworkSettings.IPAddress' > IP_ADDRESS"
+    def ipAddress = readFile('IP_ADDRESS').trim()
+    sh "rm -f IP_ADDRESS"
+    return ipAddress
+}
+
+
+node {
+    // Checkout source code from Git
+    stage 'Checkout'
+    checkout scm
+
+
+    // Build Docker image
+    stage 'Build'
+    sh "docker build -t mesosphere/cd-demo-app:${gitCommit()} ."
```

Comment:
`gitCommit()` is used several times below. Is it possible to set it at this level and re-use the variable in places below?

```
def gitCommit = gitCommit()
sh "docker build -t mesosphere/cd-demo-app:${gitCommit} ."
```